dollarskip.c

```diff
@@ -1,25 +1,63 @@
+#include <stdio.h> //perror
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h> //execvp
+#include <errno.h> //errno (error handling)
+#include <sys/wait.h> //waitpid
+
+char **parse(char *input) {
+    //will hold the command
+    char **command=malloc(sizeof(char *)*8096);
+    //the separator for strtok
+    char *separator=" ";
+    //will hold the parsed output from strtok
+    char *parsed;
+    int index=0;
+
+    //strtok returns the first word in 'input' before the space
+    parsed=strtok(input, separator);
+    //while that word isn't NULL
+    while(parsed!=NULL) {
+        command[index]=parsed; //copy it to command[index]
```

Comment:
We could use `realloc()` to grow the `command` variable if the input exceeds 8096 characters, that way the only thing that will stop crazy long commands is the amount of free memory available.
What do you say @CleanMachine1?

something like this:
(`max_size` is a `int` that starts with the value of 8096)
this is between line 20 and 21.
```c
if(index==max_size) {
        max_size += 200;
        command=realloc(command, sizeof(char *)*max_size);
}
```

Reply:
Oh, sorry I think I missed this. I mean I don't think many actually ever type a 8096 character command however it it doesn't impact the usage of the command *at all* then, Yes definitely.

Reply:
it will make it slower, however it really won't be noticeable.